app/views/contexts/section_summary_context.py

```diff
@@ -0,0 +1,154 @@
+                location.section_id
+            )
+            if repeating_title:
+                title = self._placeholder_renderer.render_placeholder(
+                    repeating_title, location.list_item_id
+                )
+        return title
+
+    def _custom_summary_elements(self, section_summary, current_location, section):
+        for summary_element in section_summary:
+            if summary_element["type"] == "List":
+                yield self._list_summary_element(
+                    summary_element, current_location, section
+                )
+
+    def _list_summary_element(
+        self, summary: Mapping, current_location, section: Mapping
+    ) -> Mapping:
+        list_collector_block = self._schema.get_list_collectors_for_list(
+            section, for_list=summary["for_list"]
+        )[0]
+
+        add_link = self._add_link(
+            summary, current_location, section, list_collector_block
+        )
+
+        list_context = ListContext(
+            self._language,
+            self._schema,
+            self._answer_store,
+            self._list_store,
+            self._progress_store,
+            self._metadata,
+        )
+
+        rendered_summary = self._placeholder_renderer.render(
+            summary, current_location.list_item_id
+        )
+
+        return {
+            "title": rendered_summary["title"],
+            "type": rendered_summary["type"],
+            "add_link": add_link,
+            "add_link_text": rendered_summary["add_link_text"],
+            "empty_list_text": rendered_summary["empty_list_text"],
+            "list_name": rendered_summary["for_list"],
+            **list_context(
+                list_collector_block["summary"],
+                for_list=list_collector_block["for_list"],
+                return_to=current_location.block_id,
+                edit_block_id=list_collector_block["edit_block"]["id"],
+                remove_block_id=list_collector_block["remove_block"]["id"],
+            ),
+        }
+
+    def _add_link(self, summary, current_location, section, list_collector_block):
+        routing_path = self._router.routing_path(
+            section["id"], current_location.list_item_id
+        )
+
+        add_link = url_for(
+            "questionnaire.block",
+            list_name=summary["for_list"],
+            block_id=list_collector_block["add_block"]["id"],
+            return_to=current_location.block_id,
+        )
+
+        if list_collector_block["id"] not in routing_path:
+            driving_question_block = QuestionnaireSchema.get_driving_question_for_list(
+                section, summary["for_list"]
+            )
+
+            if driving_question_block:
+                add_link = url_for(
+                    "questionnaire.block",
+                    block_id=driving_question_block["id"],
+                    return_to=current_location.block_id,
+                )
+        return add_link
```

Comment:
I think this is existing behaviour, but this logic is broken. If there is no driving question the url for the list collector add block (that isn't on the routing path) is returned. We could return `None` and check for it in the template.

Reply:
Yeah this emulates the current behaviour. 

I'll take a look at accessingvia the template.